src/main/java/com/github/cameltooling/dap/internal/BacklogDebuggerConnectionManager.java

```diff
@@ -70,17 +70,17 @@ public class BacklogDebuggerConnectionManager {
 	public static final String ATTACH_PARAM_PID = "attach_pid";
 	public static final String ATTACH_PARAM_JMX_URL = "attach_jmx_url";
 
-	private JMXConnector jmxConnector;
-	private MBeanServerConnection mbeanConnection;
-	private ManagedBacklogDebuggerMBean backlogDebugger;
-	private Document routesDOMDocument;
-	private IDebugProtocolClient client;
-	private Set<String> notifiedSuspendedBreakpointIds = new HashSet<>();
-	private Set<CamelThread> camelThreads = new HashSet<>();
-	int threadIdCounter = 1;
-	private Map<String, CamelBreakpoint> camelBreakpointsWithSources = new HashMap<>();
+	private volatile JMXConnector jmxConnector;
+	private volatile MBeanServerConnection mbeanConnection;
+	private volatile ManagedBacklogDebuggerMBean backlogDebugger;
+	private volatile Document routesDOMDocument;
+	private volatile IDebugProtocolClient client;
+	private final Set<String> notifiedSuspendedBreakpointIds = ConcurrentHashMap.newKeySet();
+	private final Set<CamelThread> camelThreads = ConcurrentHashMap.newKeySet();
+	private final AtomicInteger threadIdCounter = new AtomicInteger();
```

Comment:
Can we start the counter at 1? I remember that the 0 for the threadId was causing troubles in VS Code back in time. (but do not remember exactly why)

Reply:
hum, it might start with 1 as it is retrieved with incrementAndGet() here https://github.com/camel-tooling/camel-debug-adapter/pull/111/files#diff-d171417606dea4ec1810e6a4afbdbd48806b35918be3a3d774e0adfeb1e0a2ddR200

Reply:
I know, as you can see line 200, I call `threadIdCounter.incrementAndGet()`. In other words, the first I do is to increment the value, so for the first call, I will have 1